### ISSUE TITLE
 future compatibility problem in dependencies

### ISSUE THREAD

**Issue by hdhoang** [henninglive/boids-rs#1]
Hi, due to rust-lang/rust#49799, one of your dependencies will no longer build. From looking at your
Cargo.toml, we recommend you upgrade to
nalgebra version 0.14.4 or later. Thanks for your attention!

**Comment by alexxroche** [comment]
I tried updating nalgebra to version 0.14.4 in Cargo.toml but it still won't compile:
```bash
   Compiling imagefmt v4.0.0
error[E0592]: duplicate definitions with name `new`
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-0.14.4/src/core/construction.rs:510:13
    |
510 |               pub fn new($($args: N),*) -> Self {
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ duplicate definitions for `new`
...
522 | / componentwise_constructors_impl!(
523 | |     /*
524 | |      * Square matrices 1 .. 6.
525 | |      */
...   |
664 | |     U6, U1, x:(0,0), y:(1,0), z:(2,0), w:(3,0), a:(4,0), b:(5,0);
665 | | );
    | |__- in this macro invocation
    | 
   ::: ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-0.14.4/src/core/construction_slice_deprecated.rs:171:13
    |
171 |               pub fn new(data: &'a [N], $($args: usize),*) -> Self {
    |               ---------------------------------------------------- other definition for `new`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0592]: duplicate definitions with name `new`
   --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-0.14.4/src/core/construction.rs:510:13
    |
510 |               pub fn new($($args: N),*) -> Self {
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ duplicate definitions for `new`
...
522 | / componentwise_constructors_impl!(
523 | |     /*
524 | |      * Square matrices 1 .. 6.
525 | |      */
...   |
664 | |     U6, U1, x:(0,0), y:(1,0), z:(2,0), w:(3,0), a:(4,0), b:(5,0);
665 | | );
    | |__- in this macro invocation
    | 
   ::: ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-0.14.4/src/core/construction_slice_deprecated.rs:232:13
    |
232 |               pub fn new(data: &'a mut [N], $($args: usize),*) -> Self {
    |               -------------------------------------------------------- other definition for `new`
    |
    = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0592`.
error: could not compile `nalgebra`
```
but I guess that's an issue to be raised with https://github.com/dimforge/nalgebra

I even tried nalgebra 0.25.2 and got:
```bash
   Compiling boids v0.1.0 ($(pwd)/boids-rs)
error[E0412]: cannot find type `UnitQuaternionBase` in module `na::geometry`
   --> src/main.rs:16:33
    |
16  | type Quaternion = na::geometry::UnitQuaternionBase<f32, na::core::MatrixArray<f32, na::U4, na::U1>>;
    |                                 ^^^^^^^^^^^^^^^^^^ help: a type alias with a similar name exists: `UnitQuaternion`
    | 
   ::: ~/.cargo/registry/src/github.com-1ecc6299db9ec823/nalgebra-0.25.2/src/geometry/quaternion.rs:963:1
    |
963 | pub type UnitQuaternion<N> = Unit<Quaternion<N>>;
    | ------------------------------------------------- similarly named type alias `UnitQuaternion` defined here
```

This is after I've run `rustup update`
active toolchain
stable-x86_64-unknown-linux-gnu (default)
rustc 1.50.0 (cb75ad5db 2021-02-10)

**Comment by henninglive** [comment]
Fixed by upgrading to Amethyst version 0.15.3 and Nalgebra version 0.19.0. See 0758b61.
